File: Data Structures & Algorithms/sqrtx/submission-3.java
class Solution {
    public int mySqrt(int x) {
        int l = 0, r = x, res = -1;
        while(l <= r) {
            int mid = l + (r - l) / 2;
            long t = 1l * mid * mid;
            if(t <= x) {
                res = mid;
                l = mid + 1;
            }
            else r = mid - 1;
        }
        return res;
    }
}

File: Data Structures & Algorithms/anagram-groups/submission-3.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        map<string, vector<string>> mp;
        for(const string& s : strs) {
            int cnt[26] = {0};
            for(char x : s) cnt[x - 'a']++;
            string key = "";
            for(int i = 0; i < 26; i++) {
                key += to_string(cnt[i]) + ",";
            }
            mp[key].push_back(s);
        }
        vector<vector<string>> res;
        for(auto it : mp) {
            res.push_back(move(it.second));
        }
        return res;
    }
};
